src/test/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGitHubPlatformTest.java

```diff
@@ -0,0 +1,120 @@
+
+    GitHubProject project = new GitHubProject("spring-projects", "spring-integration");
+    provider.update(project, values);
+
+    assertEquals(1, values.size());
+    assertTrue(values.has(USES_GITHUB_PLATFORM));
+    assertTrue(values.of(USES_GITHUB_PLATFORM).isPresent());
+    assertFalse(values.of(USES_GITHUB_PLATFORM).get().isUnknown());
+    assertEquals(USES_GITHUB_PLATFORM.value(false), values.of(USES_GITHUB_PLATFORM).get());
+  }
+
+  @Test
+  public void testUsesGitHubProjectNull() throws IOException {
+    GitHub github = mock(GitHub.class);
+
+    UsesGitHubPlatform provider = new UsesGitHubPlatform(github);
+    provider = spy(provider);
+    when(provider.cache()).thenReturn(new GitHubProjectValueCache());
+
+    GHRepository repository = mock(GHRepository.class);
+    when(github.getRepository(any())).thenReturn(repository);
+
+    when(repository.getMirrorUrl()).thenReturn(null);
+
+    ValueHashSet values = new ValueHashSet();
+    assertEquals(0, values.size());
+
+    GitHubProject project = new GitHubProject("spring-projects", "spring-integration");
+    provider.update(project, values);
+
+    assertEquals(1, values.size());
+    assertTrue(values.has(USES_GITHUB_PLATFORM));
+    assertTrue(values.of(USES_GITHUB_PLATFORM).isPresent());
+    assertFalse(values.of(USES_GITHUB_PLATFORM).get().isUnknown());
+    assertEquals(USES_GITHUB_PLATFORM.value(true), values.of(USES_GITHUB_PLATFORM).get());
+  }
+
+  @Test
+  public void testUsesGitHubProjectEmpty() throws IOException {
+    GitHub github = mock(GitHub.class);
+    UsesGitHubPlatform provider = new UsesGitHubPlatform(github);
+    provider = spy(provider);
+    when(provider.cache()).thenReturn(new GitHubProjectValueCache());
+
+    GHRepository repository = mock(GHRepository.class);
+    when(github.getRepository(any())).thenReturn(repository);
+
+    when(repository.getMirrorUrl()).thenReturn(StringUtils.EMPTY);
+
+    ValueHashSet values = new ValueHashSet();
+    assertEquals(0, values.size());
+
+    GitHubProject project = new GitHubProject("spring-projects", "spring-integration");
+    provider.update(project, values);
+
+    assertEquals(1, values.size());
+    assertTrue(values.has(USES_GITHUB_PLATFORM));
+    assertTrue(values.of(USES_GITHUB_PLATFORM).isPresent());
+    assertFalse(values.of(USES_GITHUB_PLATFORM).get().isUnknown());
+    assertEquals(USES_GITHUB_PLATFORM.value(true), values.of(USES_GITHUB_PLATFORM).get());
+  }
+
+  @Test
+  public void testUsesGitHubProjectException() throws IOException {
+    GitHub github = mock(GitHub.class);
+
+    UsesGitHubPlatform provider = new UsesGitHubPlatform(github);
+    provider = spy(provider);
+    when(provider.cache()).thenReturn(new GitHubProjectValueCache());
+
+    when(github.getRepository(any())).thenThrow(new IOException());
+
+    ValueHashSet values = new ValueHashSet();
+    assertEquals(0, values.size());
+
+    GitHubProject project = new GitHubProject("spring-projects", "spring-integration");
```

Comment:
Please don't use a real project here. Please replace it with "org/not_existing".